tests/scripts/thread-cert/v1_2_test_multicast_registration.py

```diff
@@ -0,0 +1,254 @@
+                |
+MED_1_2 --- LEADER_1_2 --- MED_1_1
+                |
+                |
+             SED_1_1
+
+
+ 1) Bring up Leader_1_2.
+
+ 2) Bring up MED_1_2:
+    a) add MA1_LINKLOCAL which would NOT be registered in AddressRegistrationTLV of Child Update Request.
+    b) add MA2_ADMINSCOPE which would be registered in AddressRegistrationTLV of Child Update Request.
+
+ 3) Bring up SED_1_2:
+    a) add MA1_LINKLOCAL which would be registered in AddressRegistrationTLV of Child Update Request
+    b) add MA2_ADMINSCOPE which would be registered in AddressRegistrationTLV of Child Update Request.
+
+ 4) Bring up MED_1_1:
+    a) add MA1_LINKLOCAL which would NOT be registered in AddressRegistrationTLV of Child Update Request.
+    b) add MA2_ADMINSCOPE which would NOT be registered in AddressRegistrationTLV of Child Update Request.
+
+ 5) Bring up SED_1_1:
+    a) add MA1_LINKLOCAL which would be registered in AddressRegistrationTLV of Child Update Request
+    b) add MA2_ADMINSCOPE which would be registered in AddressRegistrationTLV of Child Update Request.
+
+"""
+
+
+class TestMulticastRegistration(thread_cert.TestCase):
+    topology = {
+        LEADER_1_2: {
+            'version': '1.2',
+            'whitelist': [MED_1_2, SED_1_2, MED_1_1, SED_1_1],
+        },
+        MED_1_2: {
+            'mode': 'rsn',
+            'version': '1.2',
+            'whitelist': [LEADER_1_2],
+        },
+        SED_1_2: {
+            'mode': 'sn',
+            'version': '1.2',
+            'whitelist': [LEADER_1_2],
+        },
+        MED_1_1: {
+            'mode': 'rsn',
+            'version': '1.1',
+            'whitelist': [LEADER_1_2],
+        },
+        SED_1_1: {
+            'mode': 'sn',
+            'version': '1.1',
+            'whitelist': [LEADER_1_2],
+        },
+    }
+    """All nodes are created with default configurations"""
+
+    def test(self):
+
+        # 1) Bring up Leader_1_2.
+        self.nodes[LEADER_1_2].start()
+        WAIT_TIME = WAIT_ATTACH
+        self.simulator.go(WAIT_TIME)
+        self.assertEqual(self.nodes[LEADER_1_2].get_state(), 'leader')
+
+        # 2) Bring up MED_1_2.
+        self.nodes[MED_1_2].start()
+        WAIT_TIME = WAIT_ATTACH
+        self.simulator.go(WAIT_TIME)
+        self.assertEqual(self.nodes[MED_1_2].get_state(), 'child')
+
+        # 2a) add MA1_LINKLOCAL which would NOT be registered in AddressRegistrationTLV of Child Update Request.
+        # Flush relative message queues.
+        self.flush_nodes([MED_1_2])
+        self.nodes[MED_1_2].add_ipmaddr(MA1_LINKLOCAL)
+        WAIT_TIME = WAIT_REDUNDANCE
+        self.simulator.go(WAIT_TIME)
+        messages = self.simulator.get_messages_sent_by(MED_1_2)
+        msg = messages.next_mle_message(mle.CommandType.CHILD_UPDATE_REQUEST)
+        assert not command.check_address_registration_tlv(
+            msg, MA1_LINKLOCAL
+        ), 'Error: Unexpected %s in AddressRegistrationTLV'.format(
+            MA1_LINKLOCAL)
```

Comment:
There are many code blocks like this in this script. What about we wrap in a method following logic and call the method each time for better code reuse:

**def checkMulticastReigstation(id, ...):**
- flush node messages
- add an MA
- wait for `WAIT_REDUNDANCE`
- get next `Child Update Request` message
- assert if the `AR TLV` contains or not contains the `MA`

Reply:
note that for some scenario, child update request is mandatory, but optional for the others.

Agree that there are the some code repeats.
However considering that it is just a unit test serving for the functions introduced in this PR, and per section has clear comments stating the goal, I may prefer existing way a bit as it seems to me easier for readers to know what happened per step.

Reply:
revise my previous comment. a well-named method should have no impact on the readability and understanding of the test steps. will upate.